src/common/events/events_io.cpp

```diff
@@ -306,8 +314,21 @@ error::Error TeeReader::TeeReaderLeaf::AsyncRead(
 
 void TeeReader::TeeReaderLeaf::Cancel() {
 	auto p = tee_reader_.lock();
-	assert(p);
-	p->RemoveReader(shared_from_this());
+	if (!p) {
+		// Already disconnected from the tee reader. Nothing to do.
+		return;
+	}
+
+	auto leaf_reader = p->leaf_readers_.find(shared_from_this());
+	common::io::AsyncIoHandler handler;
+	if (leaf_reader != p->leaf_readers_.end()) {
+		handler = p->leaf_readers_.find(shared_from_this())->second.pending_read.handler;
```

Comment:
Why do we need to copy it here? It will be copied also inside `CancelReader()` before erasing the leaf, isn't that one enough?

Reply:
Hmm, maybe we don't. I think I may have used it below, but then changed the approach later, and I forgot to modify this part. Let me experiment with it.

Reply:
It's not needed. Removed!